packages/web3-hooks/base/src/useFiatCurrencyRate.ts

```diff
@@ -1,12 +1,38 @@
 import { FiatCurrencyRate } from '@masknet/web3-providers'
 import { useQuery } from '@tanstack/react-query'
 import { useCurrencyType } from './useCurrencyType.js'
+import { CurrencyType } from '@masknet/web3-shared-base'
+import { pick } from 'lodash-es'
+
+const STORE_KEY = '@mask/fiat-currency-rates'
+const getData = () => {
+    const raw = localStorage.getItem(STORE_KEY)
+    if (!raw) return undefined
+    try {
+        return JSON.parse(raw) as Record<string, number>
+    } catch {
+        return undefined
+    }
+}
+
+const setData = (data: Record<string, number>) => {
+    localStorage.setItem(STORE_KEY, JSON.stringify(data))
+}
 
 export function useFiatCurrencyRate() {
     const fiatCurrencyType = useCurrencyType()
+    const currencyType = fiatCurrencyType?.toUpperCase() || CurrencyType.USD
 
     return useQuery({
-        queryKey: [fiatCurrencyType],
-        queryFn: async () => FiatCurrencyRate.getRate(fiatCurrencyType),
+        queryKey: ['fiat-currency-rates'],
+        placeholderData: getData(),
+        queryFn: async () => {
+            const allRates = await FiatCurrencyRate.getRates()
```

Comment:
Please use the API cache stead.

Reply:
Need to set cache data as `placeholderData` synchronously, to avoid rate bouncing at runtime that looks incorrect. https://mask.atlassian.net/browse/MF-5207

Reply:
> Please use the API cache stead.

The caching still exists.